resolwe_bio/tools/basespace_download.py

```diff
@@ -125,25 +149,58 @@ def get_basespace_api_file_content_url(file_id):
     return '{}/content'.format(get_basespace_api_file_url(file_id))
 
 
-def get_file_name(session, file_id, request_headers):
-    """Get file name."""
+def get_file_properties(session, file_id, request_headers):
+    """Get file name and size (in bytes)."""
     response = make_get_request(session, get_basespace_api_file_url(file_id), request_headers)
-    return response.json()['Response']['Name']
+    info = response.json()['Response']
+    return info['Name'], info['Size']
+
 
+def download_file_repeatedly(tries, session, file_id, file_name, expected_file_size, request_headers):
+    """Attempt to download BaseSpace file numerous times in case of errors."""
+    for index in range(tries):
+        try:
+            download_file(session, file_id, file_name, expected_file_size, request_headers)
+            break
+        except BaseSpaceDownloadError as error:
+            if index + 1 >= tries:
+                raise error
 
-def download_file(session, file_id, file_name, request_headers):
+        time.sleep(3)
+
+
+def download_file(session, file_id, file_name, expected_file_size, request_headers):
     """Download BaseSpace file."""
-    response = make_get_request(session, get_basespace_api_file_content_url(file_id), request_headers, True)
+    response = make_get_request(session, get_basespace_api_file_content_url(file_id), request_headers, stream=True)
 
     try:
         with open(file_name, 'wb') as f:
-            for chunk in response.iter_content(chunk_size=1024):
+            chunk_size = 1024 * 1024 * 10
+            for chunk in response.iter_content(chunk_size=chunk_size):
                 f.write(chunk)
     except FileNotFoundError:
         raise BaseSpaceDownloadError('Could not save file to {}, due to directory not being found'.format(file_name))
     except PermissionError:
         raise BaseSpaceDownloadError('Could not save file to {}, due to insufficient permissions'.format(file_name))
 
+    # Check file size.
```

Comment:
It would make sense to move validation to a separate function. And I would call it from the wrapper function.